spec/unit/provider/package/dpkg_spec.rb

```diff
@@ -52,8 +51,8 @@
   end
 
   before(:each) do
-    allow(provider).to receive(:shell_out!).with("dpkg-deb -W #{new_resource.source}", timeout: 900).and_return(dpkg_deb_status)
-    allow(provider).to receive(:shell_out).with("dpkg -s #{package}", timeout: 900).and_return(dpkg_s_status)
+    allow(provider).to receive(:shell_out!).with("dpkg-deb -W #{source}", timeout: 900).and_return(dpkg_deb_status)
```

Comment:
Yeah I think myron explains it here:

https://github.com/rspec/rspec-mocks/issues/557#issuecomment-34338814

expect statements MUST be satisfied and they are not overridable like allows are, so dropping `expect(provider).not_to receive(:shell_out!)` means it'll always blow up if you use shell_out! and any later allow/expects can't override that...